src/immer/create-slice-with-immer.ts

```diff
@@ -0,0 +1,46 @@
+import { produce } from 'immer';
+import type { Draft } from 'immer';
+
+// Utility type to infer the argument types of the actions
+type InferArgs<T> = T extends (...args: infer A) => void ? A : never;
```

Comment:
Probably, it should be this:

```suggestion
type InferArgs<T> = T extends (...args: infer A) => unknown ? A : never;
```

Reply:
In a case of a slice being created with no types provided such as:

```
const countSliceImmer = createSliceWithImmer(
  {
    name: "count",
    value: {
      countNum: 0,
    },
    actions: {
      inc: (payload) => (state) => {
        state.countNum += payload.val;
      },
      reset: () => () => {
        return { countNum: 100 }
      },
    },
  }
);
```

usage:

```
<button type="button" onClick={() => inc({val: 5 })}>
  increment
</button>
```

results in TS error:

`Argument of type '{ val: number; }' is not assignable to parameter of type 'never'.ts(2345`

Now I'm no Grandmaster of TypeScript, but in this change you do not get as much type inference if types aren't explicitly provided. TS will give errors as the payload is of type `never` if not explicitly typed. I believe createSlices operates the same way, and I don't think it's the biggest deal in the world but it is noteworthy

Reply:
Interesting. Can you add a test for this in `tests/05_createSliceWithImmer.spec.tsx`? (When you get back)

Reply:
> Interesting. Can you add a test for this in `tests/05_createSliceWithImmer.spec.tsx`? (When you get back)

I tested it and did a bit of research and believe I was mistaken in my comment. My intent was for `payload`'s type to be automatically inferred if not explicitly typed, but no other library that i know of does this including Zustand or Redux Toolkit, so it might not be the right approach.